spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java

```diff
@@ -85,8 +85,8 @@ protected UnicastingDispatcher getDispatcher() {
 	@Override
 	protected void onInit() {
 		super.onInit();
-		if (this.maxSubscribers == null) {
-			Integer max = getIntegrationProperty(IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS, Integer.class);
+		if (this.maxSubscribers == 0) {
+			int max = getIntegrationProperty(IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS, int.class);
```

Comment:
Sorry, but this is not correct.
See the behavior:
```
protected <T> T getIntegrationProperty(String key, Class<T> tClass) {
		return this.defaultConversionService.convert(this.integrationProperties.getProperty(key), tClass);
	}
```

Now imagine that `integrationProperties.getProperty()` returns `null`.
That `ConversionService` will try to convert our `null` to the expected type - `int.class` in our case.
And we end up with this:
```
private Object handleResult(@Nullable TypeDescriptor sourceType, TypeDescriptor targetType, @Nullable Object result) {
		if (result == null) {
			assertNotPrimitiveTargetType(sourceType, targetType);
		}
		return result;
	}
...
	private void assertNotPrimitiveTargetType(@Nullable TypeDescriptor sourceType, TypeDescriptor targetType) {
		if (targetType.isPrimitive()) {
			throw new ConversionFailedException(sourceType, targetType, null,
					new IllegalArgumentException("A null value cannot be assigned to a primitive type"));
		}
	}
```

so, what is the motivation for this change if it is not going to work for every use-case?

Thanks for any explanations!

Reply:
Hi @artembilan 
Yes, that's correct, I agree the conversion will fail and will throw an exception. 🙈 
I have reverted back to the previous code.

However I believe unnecessary auto unboxing is not needed:

	
	public void setMaxSubscribers(int maxSubscribers) {
		..
	}
	
and needs to be changed to 
	
	public void setMaxSubscribers(Integer maxSubscribers) {
		..
	}